tests/meson.build

```diff
@@ -35,6 +33,12 @@ pistache_test_files = [
 if get_option('PISTACHE_ENABLE_NETWORK_TESTS')
 	pistache_test_files += 'net_test'
 endif
+
+if get_option('PISTACHE_ENABLE_FLAKY_TESTS')
+    pistache_test_files += 'streaming_test'
+    pistache_test_files += 'rest_swagger_server_test'
+endif
+
```

Comment:
I would have instead suggested defining these as `test(..., suite: ['flaky'])`, so that people can select the tests at will with `meson test --no-suite flaky`. It is also possible to make `meson test` without arguments default to not running a suite:

```meson
add_test_setup(
    'noflaky',
    exclude_suites: ['flaky'],
    is_default: true,     # run this one without `meson test --setup noflaky`
)
```

Reply:
Oh that's cool, and certainly better than having to reconfigure your build to skip flaky tests.

@kiplingw for this to work in the Ubuntu PPA we would need to bump debhelper-compat to version 13, as version 12 and below run `ninja test` instead of `meson test`

Reply:
add_test_setup() works well with `ninja test`. :)

Reply:
But it would force us to skip flaky tests by default, right?

Reply:
@eli-schwartz, I think your solution is much more elegant. If you'd like to submit a PR, I'm totally fine with that.

@Tachi107, yes, I recall we'd have to bump the debhelper-compat version. I'm fine with that, but will that break all unit tests?

Reply:
> @Tachi107, yes, I recall we'd have to bump the debhelper-compat version. I'm fine with that, but will that break all unit tests?

No, but it might break compatibility with old Ubuntu releases. I don't know if the build tools of older Ubuntu versions get upgraded to support newer debhelper-compat versions. But we could at least try

Reply:
Do you want to give it a try? 

I also think the approach of flagging flaky tests as flaky in the build environment is much better than what I proposed. That way they still run, but the entire build environment doesn't break in the interim while we wait for fixes.

Reply:
> I also think the approach of flagging flaky tests as flaky in the build environment is much better than what I proposed. That way they still run, but the entire build environment doesn't break in the interim while we wait for fixes.

No, they wouldn't run, but you would be able to easily skip them with `--no-suite flaky`. An alternative that would run still run them but wouldn't mark their failure as such would be to mark them with [`should_fail: true`](https://mesonbuild.com/Reference-manual_functions.html#test).

I've submitted a PR that implements what Eli suggested as #1052

Reply:
Ok perfect. If you could add to your PR to override `override_dh_auto_test` in d/rules so that `--no-suite flaky` is added, then it should work.